dart/lib/src/sentry.dart

```diff
@@ -108,12 +108,25 @@ class Sentry {
     _hub = Hub(options);
     hub.close();
 
-    // execute integrations after hub being enabled
-    for (final integration in options.integrations) {
-      await integration(HubAdapter(), options);
+    if (_containsAppRunnerIntegration(options.integrations)) {
+      // catch any errors in Dart code running ‘outside’ the Flutter framework
+      final runZonedGuardedIntegration = RunZonedGuardedIntegration(options.integrations);
+      await runZonedGuardedIntegration(HubAdapter(), options);
+    } else {
+      final noZonedGuardedIntegration = NoZonedGuardedIntegration(options.integrations);
+      await noZonedGuardedIntegration(HubAdapter(), options);
     }
   }
 
+  static bool _containsAppRunnerIntegration(List<Integration> integrations) {
```

Comment:
shouldn't this be as simple as checking if `appRunner` is null or not? I don't see why we'd need this